crates/rethnet_eth/src/remote/cacheable_method_invocation.rs

```diff
@@ -0,0 +1,680 @@
+                CacheableMethodInvocation::GetTransactionByBlockHashAndIndex { block_hash, index },
+            ),
+            MethodInvocation::GetTransactionByBlockNumberAndIndex(block_number, index) => Ok(
+                CacheableMethodInvocation::GetTransactionByBlockNumberAndIndex {
+                    block_number,
+                    index,
+                },
+            ),
+            MethodInvocation::GetTransactionByHash(transaction_hash) => {
+                Ok(CacheableMethodInvocation::GetTransactionByHash { transaction_hash })
+            }
+            MethodInvocation::GetTransactionCount(address, block_spec) => {
+                Ok(CacheableMethodInvocation::GetTransactionCount {
+                    address,
+                    block_spec,
+                })
+            }
+            MethodInvocation::GetTransactionReceipt(transaction_hash) => {
+                Ok(CacheableMethodInvocation::GetTransactionReceipt { transaction_hash })
+            }
+            MethodInvocation::NetVersion() => Ok(CacheableMethodInvocation::NetVersion),
+
+            // Explicit to make sure if a new method is added, it is not forgotten here.
+            MethodInvocation::Accounts()
+            | MethodInvocation::BlockNumber()
+            | MethodInvocation::Call(_, _)
+            | MethodInvocation::Coinbase()
+            | MethodInvocation::EstimateGas(_, _)
+            | MethodInvocation::FeeHistory(_, _, _)
+            | MethodInvocation::GasPrice()
+            | MethodInvocation::GetFilterChanges(_)
+            | MethodInvocation::GetFilterLogs(_)
+            | MethodInvocation::Mining()
+            | MethodInvocation::NetListening()
+            | MethodInvocation::NetPeerCount()
+            | MethodInvocation::NewBlockFilter()
+            | MethodInvocation::NewFilter(_)
+            | MethodInvocation::NewPendingTransactionFilter()
+            | MethodInvocation::PendingTransactions()
+            | MethodInvocation::SendRawTransaction(_)
+            | MethodInvocation::SendTransaction(_)
+            | MethodInvocation::Sign(_, _)
+            | MethodInvocation::SignTypedDataV4(_, _)
+            | MethodInvocation::Subscribe(_)
+            | MethodInvocation::Syncing()
+            | MethodInvocation::UninstallFilter(_)
+            | MethodInvocation::Unsubscribe(_)
+            | MethodInvocation::Web3ClientVersion()
+            | MethodInvocation::Web3Sha3(_)
+            | MethodInvocation::EvmIncreaseTime(_)
+            | MethodInvocation::EvmMine(_)
+            | MethodInvocation::EvmSetAutomine(_)
+            | MethodInvocation::EvmSetNextBlockTimestamp(_)
+            | MethodInvocation::EvmSnapshot() => Err(MethodNotCacheableError(value.clone())),
+        }
+    }
+}
+
+/// Potentially cacheable Ethereum JSON-RPC method invocations for a batch call.
+#[derive(Clone, Debug)]
+#[repr(transparent)]
+pub(super) struct CacheableMethodInvocations<'a>(Vec<CacheableMethodInvocation<'a>>);
```

Comment:
I'd consider caching individual parts of the batch call separate when you introduce a resolution phase. That will allow you to get more cache hits and simplifies the cache

Reply:
Good idea, I'll try to include it 👍